Userland/Libraries/LibJS/Runtime/Temporal/ZonedDateTime.cpp

```diff
@@ -538,7 +538,29 @@ ThrowCompletionOr<NanosecondsToDaysResult> nanoseconds_to_days(VM& vm, Crypto::S
         }
     }
 
-    // 19. Return the Record { [[Days]]: days, [[Nanoseconds]]: nanoseconds, [[DayLength]]: abs(dayLengthNs) }.
+    // 19. If days < 0 and sign = 1, throw a RangeError exception.
+    if (days < 0 && sign == 1)
+        return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
```

Comment:
This error type makes no sense in this case as the nanoseconds might have come from a `Duration`, which has nothing to do with the epoch. Same below ofc.

Reply:
Sorry, that was wrong. Done now, but maybe it's too verbose, and we can combine the 3 into something like "TemporalNanosecondsToDaysConversionError".